pandas/core/generic.py

```diff
@@ -6162,9 +6163,7 @@ def fillna(
 
             if self.ndim == 1:
                 if isinstance(value, (dict, ABCSeries)):
-                    from pandas import Series
-
-                    value = Series(value)
+                    value = create_series_with_explicit_dtype(value)
```

Comment:
Can you specify the dtype here?

I find it a bit confusing to say "create series with explicit dtype" but then not specify a dtype explicitly

Reply:
Then I would need to specify `dtype_if_empty` whenever calling this function, which feels cumbersome. I feel like it's a good default behaviour to set this to `object` if not required otherwise.

Maybe we can find a better name for the function instead? What about `create_series_with_explicit_dype_if_empty`? 😄

Reply:
Personally I don't find it too cumbersome to specify the `dtype_if_empty=object` (that's more explicit (so easier to understand the code), and you need to do that for the case where it needs to be float also)

Now, in this case, I suppose it actually doesn't matter which dtype is used, so there is less value in being explicit ..;)

Reply:
Alright, I'll make it explicit everywhere then